vivo-project/VIVO
# Enable war deployment model (#197.1)

**[VIVO GitHub issue](https://github.com/vivo-project/VIVO/issues)**: https://github.com/vivo-project/VIVO/issues/3034

See details on previous reverted PR: https://github.com/vivo-project/VIVO/pull/197

# What's new?

- Moved `Vitro.defaultNamespace` and `rootUser.emailAddress` from runtime properties to context.xml.
- Maven build command no longer requires settings.xml
- Maven command no longer deploys artifact to Tomcat
- VIVO home directory provisioned on startup if not exists.

# How should this be tested?

Clone Vitro and VIVO and checkout feature branches `VIVO-3034-psuedo-rebase`
```
git clone --branch VIVO-3034-psuedo-rebase git@github.com:vivo-project/Vitro.git
git clone --branch VIVO-3034-psuedo-rebase git@github.com:vivo-project/VIVO.git
```
or
```
git clone git@github.com:vivo-project/Vitro.git
cd Vitro
checkout VIVO-3034-psuedo-rebase
cd ..
git clone git@github.com:vivo-project/VIVO.git
cd VIVO
checkout VIVO-3034-psuedo-rebase
```

Build Vitro

```
cd Vitro
mvn clean install
```

Build and run VIVO with Docker
```
cd VIVO
mvn clean package
docker-compose up
```

If not using Docker, copy the war file into Tomcat.
```
cp installer/webapp/target/vivo.war /usr/local/tomcat/webapps/ROOT.war
```

Check http://localhost:8080/

# Interested parties
@VIVO-project/vivo-committers @chenejac
